src/mixins/text_style.mixin.js

```diff
@@ -63,31 +63,42 @@
       if (!this.styles || !property || property === '') {
         return false;
       }
-      var obj = this.styles, stylesCount = 0, letterCount, foundStyle = false, style,
-          canBeSwapped = true, graphemeCount = 0;
+      var obj = this.styles, stylesCount = 0, letterCount, style,
+          allStyleObjectPropertiesMatch = true, graphemeCount = 0, styleObject;
       // eslint-disable-next-line
       for (var p1 in obj) {
         letterCount = 0;
         // eslint-disable-next-line
         for (var p2 in obj[p1]) {
+          var styleObject = obj[p1][p2],
+              stylePropertyHasBeenSet = styleObject.hasOwnProperty(property);
+
           stylesCount++;
-          if (!foundStyle) {
-            style = obj[p1][p2][property];
-            foundStyle = true;
-          }
-          else if (obj[p1][p2][property] !== style || !obj[p1][p2].hasOwnProperty(property)) {
-            canBeSwapped = false;
+
+          if (stylePropertyHasBeenSet) {
+            if (!style) {
+              style = styleObject[property];
+            }
+            else if (styleObject[property] !== style) {
+              allStyleObjectPropertiesMatch = false;
+            }
+
+            if (styleObject[property] === this[property]) {
+              delete obj[p1][p2][property];
```

Comment:
this should be a reference to styleObject, should be fine to use it to delete too.

Reply:
true!